### PR TITLE
Fix app crash when tapping profile icon

### PR DESCRIPTION
**Pull request by eVoloshchak**
### Details
Issue consists of two different problems:
1. When user just created account and logs in for the first time, we fetch all the personal details. But since concierge report is not yet created, there is no concierge's personal details, which results in app crashing when user taps on concierge's profile icon
2. When user created a new chat, `getFromReportParticipants` is called to obtain information about chat participants. If user taps on profile icon before this network request finishes, app crashes, since there is no personal details to display yet.

### Fixed Issues

https://github.com/Expensify/App/issues/8545

### Tests

1. Create a new account
2. Click on magic sign in link for Expensify you got in email
3. Set password and log into application
4. From the chat click on Concierge icon
5. Verify that app isn't crashed and personal details for concierge are displayed
6. Click `FAB` -> `New Chat`
7. Initiate a chat with a user never contacted and click the chat header _immediately_
8. Verify that app isn't crashed and personal details for the user are displayed

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Create a new account
2. Click on magic sign in link for Expensify you got in email
3. Set password and log into application
4. From the chat click on Concierge icon
5. Verify that app isn't crashed and personal details for concierge are displayed
6. Click `FAB` -> `New Chat`
7. Initiate a chat with a user never contacted and click the chat header _immediately_
8. Verify that app isn't crashed and personal details for the user are displayed

#### Web

https://user-images.githubusercontent.com/9059945/171594687-767d6450-9d1f-465f-ae7b-d3124dfe3770.mp4

#### Mobile Web

https://user-images.githubusercontent.com/9059945/171594974-3f5acf2f-e22b-4d67-8fc2-7bfab6d7723e.mp4

#### Desktop

I'm unable to launch the desktop app due to https://github.com/Expensify/App/issues/8888. It is not present if you downgrade to certain node and npm versions, but since I use MacinCloud, I'm unable to do so. So if someone with a mac could test it and upload a video, I would highly appreciate it.

#### iOS

https://user-images.githubusercontent.com/9059945/171595063-97e766b7-25d1-4450-9d27-de00410ec7ea.mp4

https://user-images.githubusercontent.com/9059945/171595128-bb3f328b-f6e9-4270-a67f-f00d4d15e4d9.mp4

#### Android

https://user-images.githubusercontent.com/9059945/171595151-ee32cd38-8658-4606-ba45-c72002462ca3.mp4

### CODE REVIEW

File: src/libs/actions/PersonalDetails.js [hunk 1]
```diff
@@ -135,6 +148,10 @@ function fetchPersonalDetails() {
             }
 
             const allPersonalDetails = formatPersonalDetails(personalDetailsList);
+            const hasConciergeDetails = _.some(allPersonalDetails, person => person.login === CONST.EMAIL.CONCIERGE);
+            if (!hasConciergeDetails) {
+                getPersonalDetailsForEmail(CONST.EMAIL.CONCIERGE);
```
Comment (marcaaron) [inline src/libs/actions/PersonalDetails.js:153]: I know we have accepted a proposal already here... but we are currently refactoring our entire API and this solution is breaking one of our cardinal rules about not chaining API commands.

Not sure how you want to handle this @thienlnam, but I don't think this mergable.

It's a backend issue because if we are fetching personal details then the details of Concierge should be delivered  with the `personalDetailsList` and we should not have to make a second API call.

File: src/pages/DetailsPage.js [hunk 2]
```diff
@@ -66,6 +67,9 @@ const getPhoneNumber = (details) => {
 
 const DetailsPage = (props) => {
     const details = props.personalDetails[props.route.params.login];
+    if (!details) {
+        return <FullScreenLoadingIndicator />;
+    }
```
Comment (marcaaron) [inline src/pages/DetailsPage.js:72]: Does seem like this would prevent the crash if we don't have the details. But it feels like a workaround that we only have to do because we don't have the details.

So the solution is to make sure we have the details - which brings us back to... it's a backend issue.

File: src/pages/home/report/ReportActionsView.js [hunk 3]
```diff
@@ -125,9 +125,10 @@ class ReportActionsView extends React.Component {
             Report.updateLastReadActionID(this.props.reportID);
         });
 
-        // If the reportID is not found then we have either not loaded this chat or the user is unable to access it.
+        // If the reportID is not found or there is no last message timestamp then
+        // we have either not loaded this chat or the user is unable to access it.
         // We will attempt to fetch it and redirect if still not accessible.
-        if (!this.props.report.reportID) {
+        if (!this.props.report.reportID || !this.props.report.lastMessageTimestamp) {
```
Comment (marcaaron) [inline src/pages/home/report/ReportActionsView.js:131]: I've never liked this existing logic because I think it's really unclear why how we save a report that has no `reportID` into Onyx. 

This change makes it even more unclear because we also have a situation where the `lastMessageTimestamp` doesn't exist. 

Can anyone answer how these reports get added and why are they missing these details?

Ultimately, it seems like the only case where we should have to "fetch" a report is if we get a Pusher update for one and it doesn't exist. I can't quite think of why else we'd need this - but also not sure how our API changes would handle this.

Basically - this is not a great design and while this crash seems bad - it's important to solve this in a way that makes sense (and probably do this internally so we can consider backend changes).

**Review by marcaaron — CHANGES_REQUESTED** [review]

**Comment by thienlnam** [comment]
Going to close this, Marc is right in that these should be solved in the backend instead of workarounds here
